### PR TITLE
feat(build): script to publish documentation to gh-pages using Travis CI

### PR DESCRIPTION
**Pull request by jvandemo**
This PR contains the code to automatically generate documentation using Travis CI (see #55).

#### Overview

Here's a general overview of how things have been implemented:

1. A commit is pushed to the repository
2. Travis CI starts a build
3. Main script runs (run unit tests)
4. If main script fails, build exits
5. `./scripts/deploy_docs_to_gh_pages.sh` script is run
6. If commit message does not contain the string `[build-docs]`, build exits
7. Output of `gulp dgeni` is committed to gh-pages branch
8. `docs/*.css` are added to `gh-pages` branch as well
9. `gh-pages` branch is pushed back to GitHub
10. GitHub updates statically hosted documentation on `angular.github.io`.

#### Documentation

Documentation has also been added in `DOCUMENTATION.md`.

#### Sample output

The following documentation is automatically generated using Travis CI:
http://jvandemo.github.io/router/

#### To do

The secure `GITHUB_TOKEN` environment variable needs to be added to `.travis.yml` since my token does not have access to the `angular/router` repository. I can help with setting this up so feel free to ask.

#### Other

This PR should fix issue #55.

### CODE REVIEW

**Comment by btford** [comment]
Thanks so much @jvandemo! I'm working to get this in today.

**Comment by jvandemo** [comment]
@btford &mdash; Thanks Brian, feel free to let me know if you believe something is missing, unclear or not working as expected.

File: .travis.yml [hunk 1]
```diff
@@ -1,22 +1,21 @@
...
-  - npm install -g gulp
-  - npm install -g karma-cli
-  - ./scripts/wait_for_browser_provider.sh
-
+- mkdir -p $LOGS_DIR
+- "./scripts/sauce_connect_setup.sh"
+- npm install
+- npm install -g gulp
+- npm install -g karma-cli
+- "./scripts/wait_for_browser_provider.sh"
```
Comment (btford) [inline .travis.yml]: Why add quotes here?

**Comment by btford** [comment]
Landed as 8d2fb77fe60225c38e6642c36705123bd771d777, d458c93ece22326f683a858cb80dead4a54e0031, and 8d2fb77fe60225c38e6642c36705123bd771d777.

I did some squashing and reformatting as needed, and added my own credentials.

Thanks so much for your contribution, @jvandemo!

File: .travis.yml (hunk 1 above)
Comment (jvandemo) [inline .travis.yml]: Not sure to be honest. I haven't added them manually. I'm wondering if the Travis CLI tool added them.

**Comment by jvandemo** [comment]
:+1:

**Pull request closed without merging**